docs/ocis/adr/0030-user-generated-client-keys.md

```diff
@@ -0,0 +1,93 @@
+---
+title: "30. User generated client keys"
+date: 2024-02-22T15:30:00+01:00
+weight: 30
+geekdocRepo: https://github.com/owncloud/ocis
+geekdocEditPath: edit/master/docs/ocis/adr
+geekdocFilePath: 0030-user-generated-client-keys.md
+---
+
+* Status: draft
+* Deciders: [@butonic](https://github.com/butonic), [@fschade](https://github.com/fschade), [@kulmann](https://github.com/kulmann), [@dragotin](https://github.com/dragotin), [@rhafer](https://github.com/rhafer)
+* Date: 2024-02-22
+
+## Context and Problem Statement
+
+Currently, the oCIS proxy is used to authenticate users based on an OpenID Connect access token or using basic auth.
+The current implementation is limited to a single identity provider, and we currently have no way to generate credentials for cli tools that do not yet support OpenID Connect
+
+The coming up calendar and contacts service requires this because most CalDAV and CardDAV clients require basic or digest authentication.
+
+## Decision Drivers <!-- optional -->
+
+## Considered Options
+
+* oCIS proxy middleware with micro store
+
+## Decision Outcome
+
+Chosen option: *???*
+
+### Positive Consequences:
+
+* TODO
+
+### Negative Consequences:
+
+* TODO
+
+## Pros and Cons of the Options <!-- optional -->
+
+### oCIS proxy middleware with micro store
+
+We use the built-in micro store to save generated access keys.
+Generated access keys will have a scope attached to limit access to oCIS services or endpoints.
+
+These scopes need to be defined - for ccs it would be sufficient to use /ccs/ endpoint.
```

Comment:
Scopes might be workt another ADR. MS graph checks scopes not only to check if a user can use an application but also to check permissions, e.g. [`Files.Read`](https://learn.microsoft.com/en-us/graph/api/drive-get?view=graph-rest-beta&tabs=http#permissions). All the scopes/permissions have a uuid tha tcan be found in the [permissions reference](https://learn.microsoft.com/en-us/graph/permissions-reference).

oCIS uses the same permission naming scheme, but only knows a fraction of what is available in ms graph: https://github.com/owncloud/ocis/blob/ee501f225e5dcc3ce9b3bf5f9ffe1f2d95e550db/services/settings/pkg/store/defaults/permissions.go.

The ms graph calendar scope I would use is [`Calendars.ReadWrite`](https://learn.microsoft.com/en-us/graph/permissions-reference#calendarsreadwrite). The constraint part is not given, so this scope allows you ar the app to read and write your own calendars.

For more background on scopes and permissions an ms graph read the [permissions overview](https://learn.microsoft.com/en-us/graph/permissions-overview?tabs=http).